src/audio.c

```diff
@@ -147,7 +152,7 @@ static void audio_callback()
     {
         /* check if next buffer is full */
         int next = (now_playing + 1) % _num_buf;
-        if (!(buf_full & (1<<next)))
+        if ((!(buf_full & (1<<next))) && !_fill_buffer_callback)
```

Comment:
At this point, you verify that we don't have a callback, but what if somebody calls pause() without one? It looks like in that case the code below will set a pause callback anyway. Is that intended behavior?